R/zchunk_L254.transportation_USA.R

```diff
@@ -87,101 +119,448 @@ module_gcam.usa_L254.transportation_USA_DISABLED <- function(command, ...) {
+      L254.StubTranTechCoef_USA
+    process_USA_to_states(L254.PerCapitaBased_trn) -> L254.PerCapitaBased_trn_USA
+    process_USA_to_states(L254.PriceElasticity_trn) -> L254.PriceElasticity_trn_USA
+    process_USA_to_states(L254.IncomeElasticity_trn) -> L254.IncomeElasticity_trn_USA
+
+    # Calibration
+    # L254.StubTranTechCalInput_USA: calibrated energy consumption by all technologies
+    L154.in_EJ_state_trn_m_sz_tech_F %>%
+      filter(year %in% BASE_YEARS) %>%
+      mutate(calibrated.value = round(value, digits = energy.DIGITS_CALOUTPUT),
+             region = state) %>%
+      left_join_error_no_match(select(UCD_techs, UCD_sector, mode, size.class, UCD_technology, UCD_fuel,
+                                      supplysector, tranSubsector, stub.technology = tranTechnology, minicam.energy.input),
+                               by = c("UCD_sector", "mode", "size.class", "UCD_technology", "UCD_fuel")) %>%
+      select(one_of(c(LEVEL2_DATA_NAMES[["StubTranTech"]])), year, minicam.energy.input, calibrated.value) ->
+      L254.StubTranTechCalInput_USA
+
+    # NOTE: NEED TO WRITE THIS OUT FOR ALL TECHNOLOGIES, NOT JUST THOSE THAT EXIST IN SOME BASE YEARS.
+    # Model may make up calibration values otherwise.
+    L254.StubTranTechCoef_USA %>%
+      filter(year %in% BASE_YEARS) %>%
+      select_if(names(L254.StubTranTechCoef_USA) %in% LEVEL2_DATA_NAMES[["StubTranTechCalInput"]]) %>%
+      left_join(L254.StubTranTechCalInput_USA,
+                by = c("region", "supplysector", "tranSubsector", "stub.technology", "year", "minicam.energy.input")) %>%
+      # Set calibration values as zero for technolgies that do not exist in some base years
+      replace_na(list(calibrated.value = 0)) %>%
+      mutate(share.weight.year = year,
+             # Create the needed variables to use the function set_subsector_shrwt
+             subsector = tranSubsector, calOutputValue = calibrated.value) %>%
+      set_subsector_shrwt() %>%
+      mutate(tech.share.weight = if_else(calibrated.value > 0, 1, 0)) %>%
+      select(one_of(c(LEVEL2_DATA_NAMES[["StubTranTechCalInput"]]))) ->
+      L254.StubTranTechCalInput_USA
+
+    # Non-motorized technologies
+    # L254.StubTranTechProd_nonmotor_USA: service output of non-motorized transportation technologies
+    L154.out_mpkm_state_trn_nonmotor_Yh %>%
+      filter(year %in% BASE_YEARS) %>%
+      mutate(calOutputValue = round(value, digits = energy.DIGITS_MPKM),
+             region = state, tranSubsector = mode) %>%
+      left_join_error_no_match(A54.globaltech_nonmotor, by = "tranSubsector") %>%
+      mutate(stub.technology = technology) %>%
+      # There is no need to match shareweights to the calOutputValue because no region should ever have a 0 here
+      select(one_of(c(LEVEL2_DATA_NAMES[["StubTranTech"]])), year, calOutputValue) ->
+      L254.StubTranTechProd_nonmotor_USA
+
+    # L254.StubTranTechCalInput_passthru_USA: calibrated input of passthrough technologies
```

Comment:
Define what the passthrough technologies are and why we need them. May need to consult with Page.

Reply:
@pkyle could you help me out here? I do not quite understand the nesting structure of the transport sector. Thanks a lot!

Reply:
The transportation structure is heavily nested! The GCAM structure of sector/subsector/technology only allows two levels of nesting within any sector, but a technology of one sector (e.g., trn_pass) can consume the output of another "sector" (e.g., trn_pass_road) that is really just used to represent lower nesting levels of that first, or parent, sector. In the transportation sector, each lower-level nesting "sector" is named by appending a string to the parent sector. So, trn_pass contains trn_pass_road which has trn_pass_road_LDV which has trn_pass_road_LDV_4W. Each of the links between any two of those sectors is done with a pass-through technology within the parent sector that consumes the output of the child sector. The technology is called a "pass-through" because it (generally) only consumes the output of the child "sector" without making any changes to it. There's an additional complication in the transportation sector, that the pass-through technologies are normal, standard GCAM technologies, not "tranTechnologies" which have different parameters read in, and perform a bunch of hard-wired unit conversions between inputs and outputs.